### PR TITLE
CI: Fix when wheels are built for staging

### LINKED ISSUE #17174: MAINT, REL: wheels not uploaded to staging on push to maintenance
Although the `1.9.2` wheels did successfully build via the new cibuildwheel infrastructure in https://github.com/scipy/scipy/actions/runs/3211264776, which is a push with `[wheel build]` to `maintenance/1.9.x` workflow, I don't see the wheels staged at either of the locations mentioned in `wheels.yml`--https://anaconda.org/multibuild-wheels-staging/scipy/files or https://anaconda.org/multibuild-wheels-staging/scipy/files.

I also have some concerns about the way the behavior is described in that file:

```
          # trigger an upload to
          # https://anaconda.org/scipy-wheels-nightly/scipy
          # for cron jobs or "Run workflow" (restricted to main branch).
          # Tags will upload to
          # https://anaconda.org/multibuild-wheels-staging/scipy
          # The tokens were originally generated at anaconda.org
```

The staging area should be used for `[wheel build]` on the maintenance branches. Tags aren't going to be used by the release manager for building or uploading wheels--the tag is only pushed up when we have the highest level of certainty that things are "ok," which is after the wheels are all built and downloaded from the staging area.

When I click on the CI status dot I also don't see any feedback about the upload process failing/skipping or whatever, which may also be useful to see long-term.

### PR DESCRIPTION
**Pull request by andyfaff**
@tylerjereddy, I hope that these changes address the issues for #17174. The changes in this PR:

Closes #17174

- cause the wheel building process to run for *every* push to a maintenance branch.
- these wheels are then uploaded to `multibuild-wheels-staging`.
- if there are multiple pushes then each push will cause a new set of wheels to be uploaded. If a push occurs before the last wheel build has finished then the CI run will abort and start again on the new push.

### CODE REVIEW

**Comment by rgommers** [comment]
> cause the wheel building process to run for _every_ push to a maintenance branch.

I think `push:` needs enabling, but it should still require `[wheel build]`. You don't want to unconditionally trigger wheel builds when there's say a one-off doc fix or a one-off fix that is being backported.

File: .cirrus.star [hunk 1]
```diff
@@ -16,6 +16,7 @@ def main(ctx):
     #   but on the cirrus-ci repo page
     # - commit message containing [wheel build]
     # - a tag that begins with v, but doesn't end with dev0
+    # - all pushes to a maintenance branch
```
Comment (tylerjereddy) [inline .cirrus.star:19]: we still want the `[wheel build]` requirement as Ralf suggests; we can remove the tag trigger above altogether I think (looks like you started doing or have done that in the code below?)

**Comment by andyfaff** [comment]
With the changes I've just made the wheels will only get built if the commit message (`git log --no-merges -1`) contains `[wheel build]`, or it's a cron job (main branch only), or started manually (subset running on github actions).

With cirrus-ci the wheel build will happen on any branch if those conditions are met.
With GHA the wheel build will only happen against pushes to the `maintenance` branches, or for PRs against `main` or `maintenance` branches, if those conditions are met.

Uploads to staging will occur if the commit message (`git log --no-merges -1`)  contains `[wheel build]` AND it's a `maintenance` branch.

File: .cirrus.star (hunk 1 above)
Comment (andyfaff) [reply .cirrus.star]: tag trigger is removed

**Review by rgommers — APPROVED** [review]
This LGTM now, thanks @andyfaff.

I'll leave it to @tylerjereddy to merge.

**Comment by tylerjereddy** [comment]
The changes look "ok" to me. The language in the PR comments above is a bit confusing re: AND vs. OR logic perhaps (i.e., necessary vs. sufficient conditions), but as far as the code changes are concerned they seem in line with what I was asking for, thanks.

We'll get a few more opportunities to give this a try soon-ish with the (at least) two release candidates for `1.10.0` and the final release.

**Comment by tylerjereddy** [comment]
I'll squash given two commits with the same message, etc.

**Comment by andyfaff** [comment]
I'll try and make it clearer in doc somewhere.

**Pull request merged**